ui/seek_bar.js

```diff
@@ -120,6 +163,97 @@ shaka.ui.SeekBar = class extends shaka.ui.RangeElement {
           this.onAdCuePointsChanged_();
         });
 
+    this.eventManager.listen(this.bar, 'mousemove', (event) => {
+      if (!this.player.getImageTracks().length) {
+        this.hideThumbnail_();
+        return;
+      }
+      const rect = this.bar.getBoundingClientRect();
+      const min = parseFloat(this.bar.min);
+      const max = parseFloat(this.bar.max);
+      // Pixels from the left of the range element
+      const mousePosition = event.clientX - rect.left;
+      // Pixels per unit value of the range element.
+      const scale = (max - min) / rect.width;
+      // Mouse position in units, which may be outside the allowed range.
+      const value = Math.round(min + scale * mousePosition);
+      // Show Thumbnail
+      this.showThumbnail_(mousePosition, value);
+    });
+
+    this.eventManager.listen(this.bar, 'keydown', (event) => {
+      this.hideThumbnailTimer_.stop();
+      const leftKeyCode = 37;
```

Comment:
Incidentally, while investigating these keyboard controls, I noticed that the bar responds to "page up" and "page down", but it does not cause a seek internally so the bar immediately jumps back to its previous position.

We should probably handle seeking via those keys, but it's out of scope for this CL. I'll write something real fast once I finish reviewing this.

Reply:
PR --> https://github.com/shaka-project/shaka-player/pull/5519